util-core/src/main/scala/com/twitter/io/ByteReader.scala

```diff
@@ -404,13 +404,13 @@ private class ByteReaderImpl(buf: Buf) extends ByteReader {
     checkRemaining(8)
     val ret =
       BigInt(buf.get(pos    ) & 0xff) << 56 |
-      BigInt(buf.get(pos + 1) & 0xff) << 48 |
-      BigInt(buf.get(pos + 2) & 0xff) << 40 |
-      BigInt(buf.get(pos + 3) & 0xff) << 32 |
-      BigInt(buf.get(pos + 4) & 0xff) << 24 |
-      BigInt(buf.get(pos + 5) & 0xff) << 16 |
-      BigInt(buf.get(pos + 6) & 0xff) <<  8 |
-      BigInt(buf.get(pos + 7) & 0xff)
+            (buf.get(pos + 1) & 0xff).toLong << 48 |
```

Comment:
I think this is still constructing many intermediate objects.  can you only OR with the bigint at the very end?